Lombiq.Tests.UI/Extensions/EmailUITestContextExtensions.cs

```diff
@@ -27,20 +27,10 @@ public static async Task GoToSmtpWebUIAsync(this UITestContext context)
         // appear, then to disappear.
         const string LoadingMaskClass = "el-loading-mask";
 
-        try
-        {
-            // We are waiting for this exact element to appear, only with one class, that indicates that the loading is
-            // happening. The loading is not always happening that's why we catch the exception, making sure that the
-            // element either did not exist, or we waited for it to appear.
-            context.CheckExistence(By.XPath($"//div[@class='{LoadingMaskClass}']"), exists: true);
-        }
-        catch (ElementNotFoundException exception)
-        {
-            context
-                .Scope.AtataContext.Log
-                .Info($"The smtp4dev site didn't reload, so the the missing loading element was ignored: " +
-                    $"{exception.Message}");
-        }
+        // We are waiting for this exact element to appear, only with one class, that indicates that the loading is
+        // happening. The loading is not always happening that's why we do it with Safely(), making sure that the
+        // element either did not exist, or we waited for it to appear.
+        context.Get(By.XPath($"//div[@class='{LoadingMaskClass}']").Safely());
```

Comment:
I also had problems here again, just for reference this is what I did: https://github.com/Lombiq/UI-Testing-Toolbox/blob/796b38f849e9da7dbca113d32c00561b5c31f9f7/Lombiq.Tests.UI/Extensions/EmailUITestContextExtensions.cs#L34-L56

Your solution seems nicer though.

Reply:
Yeah, we don't need to catch exceptions. Also, note that `Get` and friends already do retries for existence, so you don't need to retry for that outside of them.